ghcide/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -44,26 +75,82 @@ type IdeResult v = ([FileDiagnostic], Maybe v)
 -- | an IdeResult with a fingerprint
 type IdeResultNoDiagnosticsEarlyCutoff  v = (Maybe ByteString, Maybe v)
 
-ideErrorText :: NormalizedFilePath -> T.Text -> FileDiagnostic
-ideErrorText = ideErrorWithSource (Just "compiler") (Just DiagnosticSeverity_Error)
+ideErrorText :: NormalizedFilePath -> T.Text -> Maybe (MsgEnvelope GhcMessage) -> FileDiagnostic
+ideErrorText fdFilePath msg origMsg =
+  ideErrorWithSource (Just "compiler") (Just DiagnosticSeverity_Error) fdFilePath msg origMsg
+
+ideErrorFromLspDiag
+  :: LSP.Diagnostic
+  -> NormalizedFilePath
+  -> Maybe (MsgEnvelope GhcMessage)
+  -> FileDiagnostic
+ideErrorFromLspDiag lspDiag fdFilePath mbOrigMsg =
+  let fdShouldShowDiagnostic = ShowDiag
+      fdLspDiagnostic =
+        lspDiag
+          & attachReason (fmap (diagnosticReason . errMsgDiagnostic) mbOrigMsg)
+          & setGhcCode mbOrigMsg
```

Comment:
I think so, yes.

Reply:
I was actually looking at when this function is called with a `Just` and couldn't easily find anything, because there are so many callers that pass along a value for which you then have to transitively search.

Reply:
OK, I found this now:

* https://github.com/haskell/haskell-language-server/pull/4433/files/0776c65490d8840d03ec8740f0c8048ea1d55b35#diff-9ec5d8d4031c00a406a3c0ce65ca9e155f3a05d7014a97469f89b5fd3bb65bb9R71